config/validator.js

```diff
@@ -148,6 +148,10 @@ function validateConfig(configOrFileName) {
     // core
     expectedKeys.push('core');
     assertBoolean('config.core.enableCustomConstraints', config.core.enableCustomConstraints);
+    assertNumber('config.core.overlayShardSize', config.core.overlayShardSize);
+    if(config.core.overlayShardSize < 1000){
+        throw new Error('Overlay shard size must be at least 1000.');
+    }
```

Comment:
Mainly it is there not to set a negative or zero number, but then I thought probably setting it too low would do more harm at the end for the user.
The tests are not validating whenever they change these values.